frontend/ui/elements/screensaver_menu.lua

```diff
@@ -153,12 +154,24 @@ return {
                         end,
                     },
                     {
-                        text = _("Stretch covers and images to fit screen"),
+                        text_func = function()
+                            if G_reader_settings:nilOrFalse("screensaver_stretch_images") then
+                                return _("Stretch to fit screen")
+                            elseif G_reader_settings:readSetting("screensaver_stretch_limit_percentage") then
+                                return T(_("Stretch to fit screen: limit to %1%"),
+                                    G_reader_settings:readSetting("screensaver_stretch_limit_percentage"))
+                            else
+                                return _("Stretch to fit screen: unlimited")
+                            end
+                        end,
                         checked_func = function()
                             return G_reader_settings:isTrue("screensaver_stretch_images")
                         end,
-                        callback = function()
-                            G_reader_settings:toggle("screensaver_stretch_images")
+                        help_text_func = function()
+                            return T(_("If the image and the screen have a similar aspect ratio (±%1%), stretch the image instead of keeping its aspect ratio."), G_reader_settings:readSetting("screensaver_stretch_limit_percentage"))
```

Comment:
Bad help_text when the setting for %1 does not exist.
As you show everything in the menu item text, no need to complicate the help text and show the % in it, it can just be static text.